src/textviewwidget.cpp

```diff
@@ -0,0 +1,110 @@
+#include "textviewwidget.h"
+
+#include <algorithm>
+
+#include "utils.h"
+
+namespace newsboat {
+
+TextviewWidget::TextviewWidget(const std::string& textview_name,
+	std::shared_ptr<Stfl::Form> form)
+	: textview_name(textview_name)
+	, form(form)
+	, num_lines(0)
+{
+}
+
+void TextviewWidget::set_lines(uint32_t number_of_lines)
+{
+	num_lines = number_of_lines;
+}
+
+void TextviewWidget::scroll_up()
+{
+	uint32_t offset = get_scroll_offset();
```

Comment:
I believe this and the rest of local variables in this file can be marked `const`.